modules/sara_r4/cellular_r4_api.c

```diff
@@ -2825,3 +2828,51 @@ CellularError_t Cellular_Init( CellularHandle_t * pCellularHandle,
 }
 
 /*-----------------------------------------------------------*/
+
+/* FreeRTOS Cellular Library API. */
+/* coverity[misra_c_2012_rule_8_7_violation] */
+CellularError_t Cellular_SocketSetSockOpt( CellularHandle_t cellularHandle,
+                                           CellularSocketHandle_t socketHandle,
+                                           CellularSocketOptionLevel_t optionLevel,
+                                           CellularSocketOption_t option,
+                                           const uint8_t * pOptionValue,
+                                           uint32_t optionValueLength )
+{
+    CellularError_t err = CELLULAR_SUCCESS;
+
+    err = _Cellular_isSockOptSupport( optionLevel, option );
+
+    if( err == CELLULAR_SUCCESS )
+    {
+        err = Cellular_CommonSocketSetSockOpt( cellularHandle, socketHandle, optionLevel, option,
+                                               pOptionValue, optionValueLength );
+    }
+
+    return err;
+}
+
+/*-----------------------------------------------------------*/
+
+static CellularError_t _Cellular_isSockOptSupport( CellularSocketOptionLevel_t optionLevel,
```

Comment:
Static function should be placed before public function.

Reply:
Moved.